tests/socket.rs

```diff
@@ -1297,3 +1297,39 @@ fn header_included() {
     let got = socket.header_included().expect("failed to get value");
     assert_eq!(got, true, "set and get values differ");
 }
+
+#[test]
+#[cfg(all(
+    feature = "all",
+    any(target_os = "android", target_os = "fuchsia", target_os = "linux")
+))]
+fn original_dst() {
+    let socket = Socket::new(Domain::IPV4, Type::STREAM, None).unwrap();
+    match socket.original_dst() {
+        Ok(_) => panic!("original_dst on non-redirected socket should fail"),
```

Comment:
How difficult would it be to add a test that successfully retrieved the address?

Reply:
You'd need to configure `iptables` redirect in the operating system, which seems invasive for a test of the bindings.